app/README.md

```diff
@@ -0,0 +1,246 @@
+
+> **Important!** It is a given that as part of the refactoring unit tests are either updated or added to cover the changes in behaviour
+
+### Controllers
+
+Our intent is that the controller does not change at all. It should continue to call the generic service, for example, `ViewTransactionChargeService`.
+
+```javascript
+  static async view (req, h) {
+    const result = await ViewTransactionChargeService.go(req.app.billRun, req.params.transactionId)
+
+    return h.response(result).code(200)
+  }
+```
+
+### Services
+
+This is where we expect most refactoring to happen. The key change is that the role of the main service should change from handling the request to determining which ruleset based service should.
+
+#### Main service
+
+Imagine the service before refactoring looked like this.
+
+```javascript
+class ViewTransactionChargeService {
+  static async go (billRun, transactionId) {
+    let transaction = await this._transaction(transactionId)
+
+    return this._response(transaction)
+  }
+
+  static _transaction (transactionId) {
+    return TransactionModel.query().findById(transactionId)
+  }
+
+  static _response (transaction) {
+    const presenter = new ViewTransactionChargePresenter(transaction)
+
+    return presenter.go()
+  }
+}
+```
+
+The contents should first be copy & pasted into 2 new services; `ViewTransactionChargePresrocService` and `ViewTransactionChargeSrocService`. Then the service should be refactored to become the 'router' i.e. the thing which determines which of the ruleset based services to call when a request comes in.
+
+```javascript
+class ViewTransactionChargeService {
+  static async go (billRun, transactionId) {
+    switch (billRun.ruleset) {
+      case 'sroc':
+        return ViewTransactionChargeSrocService.go(billRun, transactionId)
+      case 'presroc':
+        return ViewTransactionChargePresrocService.go(billRun, transactionId)
+      default:
+        throw Error('Unrecognised ruleset')
+    }
+  }
+}
+```
+
+#### Ruleset services
+
+With 2 ruleset based services we are now free to refactor as required. In our example the key difference will be which presenter is used to generate the response (see the [Translators and Presenters](#transalators-and-presenters) for details about the presenter refactoring).
+
+```javascript
+class ViewTransactionChargeSrocService {
+  static async go (billRun, transactionId) {
+    let transaction = await this._transaction(transactionId)
+
+    return this._response(transaction)
+  }
+
+  static _transaction (transactionId) {
+    return TransactionModel.query().findById(transactionId)
+  }
+
+  static _response (transaction) {
+    const presenter = new ViewTransactionChargeSrocPresenter(transaction) # <- This line!
```

Comment:
I wonder if it would be better to have `# <- This line!` be a valid comment instead:
```suggestion
    const presenter = new ViewTransactionChargeSrocPresenter(transaction) // <- This line!
```